### ISSUE TITLE
New RGCN model in Link Prediction needs much more memory

### ISSUE THREAD

**Issue by JulianSe26** [dmlc/dgl#3811]
https://github.com/dmlc/dgl/blob/e0f0fa2a788dea67078a06ae5883d1e23bc4c878/examples/pytorch/rgcn/model.py#L25

Hi there, 

thanks for your work. I noticed that when I adopted the new implemention of the RGCN in link prediction, the memory consumption increased rapidly. Is there any advatage using the new "version" of RGCN ? Were there bugs in the old implementation ? 

Thanks in advance

**Comment by jermainewang** [comment]
Hi, we did change the example in the latest release. The main goal is to make it shorter, cleaner and easier to understand for beginners. As a result, we also simplify the model architecture as follows:
* Use `torch.nn.Embedding` to represent the trainable node embeddings which is slightly different than the original code where the first layer accepts an integer node ID tensor and the node embeddings are implicitly encoded as the layer weight.
* Use one optimizer for both node embedding and model weights. Previously, there is a dedicated sparse optimizer for node embedding.

In summary, the model architecture of the new example is slightly different than the previous one as well as the optimizer setup, which may cause more memory usage. In the future, we plan to bring back the previous implementation by a collection of advanced/optimized examples.
